proof_of_concept/asset.py

```diff
@@ -0,0 +1,51 @@
+"""Classes for describing assets."""
+from typing import Any, Dict
+
+
+class Asset:
+    """Asset, a representation of a computation or piece of data."""
+
+    def __init__(self, name: str, data: Any, metadata: Any):
+        """Constructor.
+
+        TODO: typehint Metadata class for metadata argument, problematic
+            due to circular imports
+        Args:
+            name: Name of the asset
+            data: Data related to the asset
+            metadata: Metadata related to the asset
+        """
+        self.name = name
```

Comment:
I am curious whether these attributes are what you have in mind @LourensVeen

Reply:
The data would eventually be a reference to a container image containing the data, I think it now just contains an arbitrary Python object.

I think there's Metadata being used in the AssetStore. (Intermediate) results are `DataAsset`s, and they need Metadata for provenance and also for security, as what you can do with that intermediate result depends on how it was calculated.